packages/internal/src/build/babel/common.ts

```diff
@@ -62,3 +68,65 @@ export const getCommonPlugins = () => {
     ['@babel/plugin-proposal-private-property-in-object', { loose: true }],
   ]
 }
+
+/**
+ * Finds, reads and parses the [ts|js]config.json file
+ * @returns The config object
+ */
+export const parseConfigFiles = () => {
+  const rwPaths = getPaths()
+
+  const parseConfigFile = (basePath: string) => {
+    let configPath = path.join(basePath, 'tsconfig.json')
+    if (!fs.existsSync(configPath)) {
+      configPath = path.join(basePath, 'jsconfig.json')
+      if (!fs.existsSync(configPath)) {
+        return null
+      }
+    }
+    return parseConfigFileTextToJson(
+      configPath,
+      fs.readFileSync(configPath, 'utf-8')
+    )
+  }
+  const apiConfig = parseConfigFile(rwPaths.api.base)
+  const webConfig = parseConfigFile(rwPaths.web.base)
+
+  return {
+    api: apiConfig?.config ?? null,
+    web: webConfig?.config ?? null,
+  }
+}
+
+/**
+ * Extracts and formats the paths from the [ts|js]config.json file
+ * @param config The config object
+ * @returns {Record<string, string>} The paths object
+ */
+export const getPathsFromConfig = (config: {
+  compilerOptions: { baseUrl: string; paths: string }
+}): Record<string, string> => {
+  if (!config) {
+    return {}
+  }
+
+  const { baseUrl, paths } = config.compilerOptions
+  const pathsObj: Record<string, string> = {}
+  for (const [key, value] of Object.entries(paths)) {
```

Comment:
Question: do we need to do this? Or can we just remove the predefined stuff in: https://github.com/redwoodjs/redwood/pull/7575/files#diff-9aac922074f3addea47d2d5cddf7eb2fa22adc679d08061f8aab28581bd14430R113

Reply:
not sure what you mean by removing the predefined stuff there, since we don't remove them, we leave them as they are and exclude them from the transformation, otherwise quite a few things break.

Reply:
👍